File: src/lessons/welcome/variables/VariablesEntity.py
def forward(i=1):
    if i>1:
        errorMsg("Sorry Dave, I cannot let you use forward(i>1) in this exercise")
    entity.forward()
def backward(i=1):
    if i>1:
        errorMsg("Sorry Dave, I cannot let you use backward(i>1) in this exercise")
    entity.backward()

# BEGIN SOLUTION 
cpt = 0
while not isOverBaggle():
  cpt += 1 
  forward()
pickUpBaggle()
while cpt>0:
  backward()
  cpt -= 1
dropBaggle()
# END SOLUTION

File: src/lessons/welcome/variables/VariablesEntity.java
package lessons.welcome.variables;

import plm.core.model.Game;

public class VariablesEntity extends plm.universe.bugglequest.SimpleBuggle {
	@Override
	public void forward(int i)  { 
		throw new RuntimeException(Game.i18n.tr("Sorry Dave, I cannot let you use forward with an argument in this exercise. Use a loop instead."));
	}

	@Override
	public void backward(int i) {
		throw new RuntimeException(Game.i18n.tr("Sorry Dave, I cannot let you use backward with an argument in this exercise. Use a loop instead."));
	}


	@Override
	public void run() { 
		/* BEGIN SOLUTION */
		int cpt = 0;
		while (!isOverBaggle()) {
			cpt++;
			forward();
		}
		pickupBaggle();
		while (cpt>0) {
			backward();
			cpt--;
		}
		dropBaggle();
		/* END SOLUTION */
	}
}
